### DISCUSSION TITLE
AsyncNotifierProviderFamily (AsyncNotifierProvider.family) example

### DISCUSSION THREAD

**Discussion by felipe-xavier** [rrousselGit/riverpod#1813] [Q&A] [answered]
I couldn't find examples from the documentation or any where else.

How to make the following provider with family:

```Dart
final provider = AsyncNotifierProvider<MyNotifier, String>(MyNotifier.new);
```

So that the class could use the attribute `final int value`:

```Dart

class MyNotifier extends AsyncNotifier<String> {
  final int value;
  @override
  FutureOr<String> build() async {
    await Future.delayed(const Duration(seconds: value));
    return "BUILD DONE";
  }
}

```

**Comment by rrousselGit** [comment, marked as answer]
```dart
final provider = AsyncNotifierProvider.autoDispose.family<MyNotifier, String, int>(MyNotifier.new);

class MyNotifier extends FamilyAsyncNotifier<String> {
  @override
  FutureOr<String> build(int value) async {

  }
}

```

**Reply by Zekfad** [reply]
@rrousselGit is it named `FamilyAsyncNotifier` / `AutoDisposeFamilyAsyncNotifier` now or am I something missing?

**Reply by rrousselGit** [reply]
You're correct.

**Reply by scopendo** [reply]
I think there's a mistake in the example above – it omits the argument type.  It should be:

```dart
class MyNotifier extends FamilyAsyncNotifier<String, int> {
  ...
}
```

**Reply by tonnyavery** [reply]
How would one access family parameter outside of `build` method?

**Reply by tonnyavery** [reply]
Also, how to use `select` and `notifier` if AsyncNotifierProvider has Family? I get that those things are not defined for `AsyncNotifierProviderFamily`

**Reply by scopendo** [reply]
> How would one access family parameter outside of `build` method?

The parameter passed to the `build` method is used to load the current state – if you need the value then either store it in the state or add a private variable to the notifier.

**Reply by tonnyavery** [reply]
Thanks, managed to figure it out.
Riverpod looks like an interesting idea and approach, but needs much more thorough documentation, examples and tutorials.

**Comment by felipe-xavier** [comment]
dear @rrousselGit , if I change my provider from `AsyncNotifierProviderFamily` to `AsyncNotifierProvider.family.autoDispose` I get this error:
`Error: Type argument 'MyNotifier' doesn't conform to the bound 'AutoDisposeFamilyAsyncNotifier<T, Arg>' of the type variable 'NotifierT' on 'AutoDisposeAsyncNotifierProviderFamilyBuilder.call`

**Reply by rrousselGit** [reply]
Subclass AutoDisposeAsyncNotifierFamily then

**Reply by scopendo** [reply]
> AutoDisposeFamilyAsyncNotifier
